backend/src/test/java/com/google/sps/ParticipantDatastoreTest.java

```diff
@@ -0,0 +1,219 @@
+    assertThat((long) entityB.getProperty(PROPERTY_MATCHID)).isEqualTo(MATCHID_DEFAULT);
+    assertThat((long) entityB.getProperty(PROPERTY_TIMESTAMP)).isEqualTo(TIMESTAMP_DEFAULT);
+  }
+
+  @Test
+  public void addGetOneParticipant() throws EntityNotFoundException {
+    // Add one participant to datastore and get participant
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
+    Participant participant =
+        new Participant(
+            PERSON_A,
+            STARTTIMEAVAILABLE_DEFAULT,
+            ENDTIMEAVAILABLE_DEFAULT,
+            DURATION_DEFAULT,
+            MATCHID_DEFAULT,
+            TIMESTAMP_DEFAULT);
+
+    participantDatastore.addParticipant(participant);
+
+    Participant participantFromUsername = participantDatastore.getParticipantFromUsername(PERSON_A);
+    assertThat(participantFromUsername.getUsername()).isEqualTo(PERSON_A);
+    assertThat(participantFromUsername.getStartTimeAvailable())
+        .isEqualTo(STARTTIMEAVAILABLE_DEFAULT);
+    assertThat(participantFromUsername.getEndTimeAvailable()).isEqualTo(ENDTIMEAVAILABLE_DEFAULT);
+    assertThat(participantFromUsername.getDuration()).isEqualTo(DURATION_DEFAULT);
+    assertThat(participantFromUsername.getMatchId()).isEqualTo(MATCHID_DEFAULT);
+    assertThat(participantFromUsername.getTimestamp()).isEqualTo(TIMESTAMP_DEFAULT);
+  }
+
+  @Test
+  public void getNonexistentParticipant() {
+    // Try to get participant from username that's not in datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
+    Participant participant =
+        new Participant(
+            PERSON_A,
+            STARTTIMEAVAILABLE_DEFAULT,
+            ENDTIMEAVAILABLE_DEFAULT,
+            DURATION_DEFAULT,
+            MATCHID_DEFAULT,
+            TIMESTAMP_DEFAULT);
+
+    Participant participantFromUsername = participantDatastore.getParticipantFromUsername(PERSON_A);
+    assertThat(participantFromUsername).isNull();
+  }
+
+  @Test
+  public void removeParticipant() {
+    // Add and then remove participant from datastore
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    ParticipantDatastore participantDatastore = new ParticipantDatastore(datastore);
+    Participant participant =
+        new Participant(
+            PERSON_A,
+            STARTTIMEAVAILABLE_DEFAULT,
+            ENDTIMEAVAILABLE_DEFAULT,
+            DURATION_DEFAULT,
+            MATCHID_DEFAULT,
+            TIMESTAMP_DEFAULT);
+
+    participantDatastore.addParticipant(participant);
+    participantDatastore.removeParticipant(participant.getUsername());
+
+    Participant participantFromUsername = participantDatastore.getParticipantFromUsername(PERSON_A);
+    assertThat(participantFromUsername).isNull();
```

Comment:
NIt: Do you think you should add test-cases with multiple partcipants and potentially comparing the lengths

Reply:
What do you mean? Lengths of what?